pkg/snapshot/restorer/restorer_test.go

```diff
@@ -665,290 +665,338 @@ var _ = Describe("Running Restorer", func() {
 		})
+	})
 
-		Context("With first few snapshots in v1 dir and some more incr snapshots are in v2 dir", func() {
-			It("should restore from v1 dir and v2 dir", func() {
-				memberPath := path.Join(etcdDir, "member")
+	Context("With first few snapshots in v1 dir and some more incr snapshots are in v2 dir", func() {
+		It("should restore from v1 dir and v2 dir", func() {
+			//Take snapshots for v1 dir
+			compressionConfig := compressor.NewCompressorConfig()
+			ctx, cancel := context.WithTimeout(testCtx, time.Duration(2*time.Second))
+			resp := &utils.EtcdDataPopulationResponse{}
+			err = takeValidV1Snaps(logger, cmpctStoreDir, resp, deltaSnapshotPeriod, ep, ctx.Done(), compressionConfig, fullSnapInV1)
+			cancel()
 
-				//Take snapshots for v1 dir
-				compressionConfig := compressor.NewCompressorConfig()
-				ctx, cancel := context.WithTimeout(testCtx, time.Duration(2*time.Second))
-				err = takeValidV1Snaps(logger, snapstoreDir, deltaSnapshotPeriod, endpoints, ctx.Done(), compressionConfig, fullSnapInV1)
-				cancel()
+			baseSnapshot, deltaSnapList, err = miscellaneous.GetLatestFullSnapshotAndDeltaSnapList(store)
+			Expect(err).ShouldNot(HaveOccurred())
 
-				// remove the member dir
-				err = os.RemoveAll(memberPath)
-				Expect(err).ShouldNot(HaveOccurred())
+			rstr = NewRestorer(store, logger)
 
-				baseSnapshot, deltaSnapList, err = miscellaneous.GetLatestFullSnapshotAndDeltaSnapList(store)
-				Expect(err).ShouldNot(HaveOccurred())
+			restoreOpts := brtypes.RestoreOptions{
+				Config:        restorationConfig,
+				BaseSnapshot:  baseSnapshot,
+				DeltaSnapList: deltaSnapList,
+				ClusterURLs:   clusterUrlsMap,
+				PeerURLs:      peerUrls,
+			}
 
-				rstr = NewRestorer(store, logger)
+			//Restore
 
-				restoreOpts := brtypes.RestoreOptions{
-					Config:        restorationConfig,
-					BaseSnapshot:  baseSnapshot,
-					DeltaSnapList: deltaSnapList,
-					ClusterURLs:   clusterUrlsMap,
-					PeerURLs:      peerUrls,
-				}
+			// remove the member dir
+			err = os.RemoveAll(path.Join(emDir, "member"))
+			Expect(err).ShouldNot(HaveOccurred())
 
-				//Restore
-				err = rstr.RestoreAndStopEtcd(restoreOpts)
-				Expect(err).ShouldNot(HaveOccurred())
-				err = utils.CheckDataConsistency(testCtx, restoreOpts.Config.RestoreDataDir, 400, logger)
-				Expect(err).ShouldNot(HaveOccurred())
-			})
+			err = rstr.RestoreAndStopEtcd(restoreOpts)
+			Expect(err).ShouldNot(HaveOccurred())
+			err = utils.CheckDataConsistency(testCtx, restoreOpts.Config.RestoreDataDir, resp.KeyTo, logger)
+			Expect(err).ShouldNot(HaveOccurred())
 		})
+	})
 
-		Context("With first few snapshots in v2 dir and some more incr snapshots are in v1 dir", func() {
-			It("should restore from v1 dir and v2 dir", func() {
-				memberPath := path.Join(etcdDir, "member")
+	Context("With first few snapshots in v2 dir and some more incr snapshots are in v1 dir", func() {
+		It("should restore from v1 dir and v2 dir", func() {
```

Comment:
Shouldn't you also be taking some v2 snapshots first in this spec before taking v1 snapshots? I see that only v1 snapstore is being populated before checking for restoration.